SwedbankPaySDK/Classes/Merchant Backend/Problems.swift

```diff
@@ -0,0 +1,185 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import Alamofire
+
+enum Problems {
+    static func parseProblem(
+        response: AFDataResponse<Data>
+    ) -> SwedbankPaySDK.Problem {
+        do {
+            return try parseProblemOrFail(response: response)
+        } catch {
+            return makeUnexpectedContentProblem(response: response)
+        }
+    }
+    
+    static func makeUnexpectedContentProblem(
+        response: AFDataResponse<Data>
+    ) -> SwedbankPaySDK.Problem {
+        // we should always have response here, but we need some value
+        let status = response.response?.statusCode ?? -1
+        let isClient = (400...499).contains(status)
+        if isClient {
+            return .client(.unexpectedContent(
+                status: status,
+                contentType: response.response?.mimeType,
+                body: response.data
+            ))
+        } else {
+            return .server(.unexpectedContent(
+                status: status,
+                contentType: response.response?.mimeType,
+                body: response.data
+            ))
+        }
+    }
+}
+
+private typealias JsonObject = [String: Any]
+
+private extension JsonObject {
+    var type: String { self["type"] as? String ?? "about:blank" }
+    var title: String? { self["title"] as? String }
+    var status: Int? { self["status"] as? Int }
+    var detail: String? { self["detail"] as? String }
+    var instance: String? { self["instance"] as? String }
+}
+
+
+private func parseProblemOrFail(
+    response: AFDataResponse<Data>
+) throws -> SwedbankPaySDK.Problem {
+    let data = try requireSome(response.data)
+    let jsonValue = try JSONSerialization.jsonObject(with: data)
+    let json = try requireSome(jsonValue as? JsonObject)
+    let status = try json.status ?? requireSome(response.response).statusCode
+    let problemSpace = try getProblemSpace(status: status)
+    return problemSpace.parseProblem(status: status, json: json)
+}
+
+private func getProblemSpace(status: Int) throws -> ProblemSpace {
+    switch status {
+    case 400...499: return clientProblems
+    case 500...599: return serverProblems
+    default: throw PreconditionError.instance
+    }
+}
+
+private struct ProblemSpace {
+    private let parseProblemOfType: (String, Int, JsonObject) -> SwedbankPaySDK.Problem
+    
+    init<T>(
+        problemType: @escaping (T) -> SwedbankPaySDK.Problem,
+        unknownProblemType: @escaping (String, String?, Int, String?, String?, [String: Any]) -> T,
+        parseKnownProblem: @escaping (String, Int, JsonObject) -> T?
+    ) {
+        parseProblemOfType = { type, status, json in
+            let t = parseKnownProblem(type, status, json)
```

Comment:
Codacy found an issue: [Variable name should be between 3 and 40 characters long: 't'](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-ios/pullRequest?prid=6260585)